pkg/devfile/generator/utils_test.go

```diff
@@ -663,7 +679,7 @@ func TestGetServiceSpec(t *testing.T) {
 		wantPorts           []corev1.ServicePort
 	}{
 		{
-			name: "multiple endpoints share the same port",
+			name: "multiple endpoints have different ports",
```

Comment:
This test should cover most of our scenarios since we have validation on unique ports but should we handle the exception case as well where `dedicatedPod=true`? https://github.com/devfile/api/blob/main/pkg/validation/validation-rule.md#endpoints

Reply:
we do not specifially do tests for `deficatedPod` in library, since it is currently not supported in library. see https://github.com/devfile/api/issues/670